mopidy/core/tracklist.py

```diff
@@ -579,7 +595,9 @@ def _mark_unplayable(self, tl_track):
     def _mark_played(self, tl_track):
         """Internal method for :class:`mopidy.core.PlaybackController`."""
         if self.get_consume() and tl_track is not None:
-            self.remove({"tlid": [tl_track.tlid]})
+            logger.debug("Consuming Track %d", tl_track.tlid)
+            tl_tracks = self.filter({"tlid": [tl_track.tlid]})
+            self._remove_tracks(tl_tracks, stop_if_current=False)
```

Comment:
How about
```
self._remove_tracks([tl_track], stop_if_current=False)
```

Reply:
Ha yes, of course. I just used .filter() without thinking about what it does.